rules/alignment.smk

```diff
@@ -207,6 +207,72 @@ elif config['aligner'] == 'salmon':
             """
 
 
+elif config['aligner'] == 'star':
+    rule star_index:
+        input:
+            genome = expand("{genome_dir}/{{assembly}}/{{assembly}}.fa", **config),
+            gtf = expand("{genome_dir}/{{assembly}}/{{assembly}}.gtf", **config)
+        output:
+            dir = directory(expand("{genome_dir}/{{assembly}}/index/{aligner}", **config)),
+            tmp = temp(directory(expand("{genome_dir}/{{assembly}}/index/{aligner}_tmp", **config)))
+        log:
+            default = expand("{log_dir}/{aligner}_index/{{assembly}}.log", **config),
+            star = expand("{log_dir}/{aligner}_index/{{assembly}}_Log.out", **config)
+        benchmark:
+            expand("{benchmark_dir}/{aligner}_index/{{assembly}}.benchmark.txt", **config)[0]
+        params:
+            config['star_index']
+        threads: 20
+        conda:
+            "../envs/star.yaml"
+        shell:
+            """
+            mkdir {output.dir}
+            mkdir {output.tmp}
+            
+            STAR --runMode genomeGenerate --genomeFastaFiles {input.genome} --sjdbGTFfile {input.gtf} --genomeDir {output.dir} \
+            --runThreadN {threads} --outFileNamePrefix {output.tmp}/ {params} > {log.default} 2>&1
+            
+            # STAR also creates an extended log.
+            if [ -f {output.tmp}/Log.out ]; then
+                mv {output.tmp}/Log.out {log.star}
+            fi
+            """
+
+
+    rule star_quant:
+        input:
+            reads=get_reads,
+            index=expand("{genome_dir}/{{assembly}}/index/{aligner}", **config)
+        output:
+            dir=temp(directory(expand("{result_dir}/{aligner}/{{assembly}}-{{sample}}", **config))),
+            pipe=get_alignment_pipes()
+        log:
+            directory(expand("{log_dir}/{aligner}_align/{{assembly}}-{{sample}}", **config))
```

Comment:
Should this be a directory?

Reply:
Star creates a set of logs (5 files, each with different details), some of which are several MB large. 

By keeping all log files separate the experience is the same as for users of stand-alone star.
By storing all log files within the log/star_align directory, the file tree is not cluttered.